Sources/Target/Darwin/Thread.cpp

```diff
@@ -33,7 +33,88 @@ namespace Darwin {
 using Host::Darwin::LibProc;
 
 Thread::Thread(ds2::Target::Process *process, ThreadId tid)
-    : super(process, tid), _lastSyscallNumber(-1) {}
+    : super(process, tid), _lastSyscallNumber(-1) {
+  //
+  // Initially the thread is stopped.
+  //
+  _resumeWithPtrace = true;
+}
+
+ErrorCode Thread::suspend() {
+  ErrorCode error = kSuccess;
+  if (_state == kRunning) {
+    error = process()->mach().suspend(ProcessThreadId(process()->pid(), tid()));
+    if (error != kSuccess)
+      return error;
+    updateStopInfo(1407);
+  }
+
+  if (_state == kTerminated) {
+    error = kErrorProcessNotFound;
+  }
+
+  return error;
+}
+
+ErrorCode Thread::step(int signal, Address const &address) {
+  ErrorCode error = kSuccess;
+  if (_state == kStopped || _state == kStepped) {
+    DS2LOG(Debug, "stepping tid %d", tid());
+  } else if (_state == kTerminated) {
+    error = kErrorProcessNotFound;
+  }
+  return error;
+}
+
+ErrorCode Thread::resume(int signal, Address const &address) {
+  ErrorCode error = kSuccess;
+  if (_state == kStopped || _state == kStepped) {
+    if (signal == 0) {
+      switch (_stopInfo.signal) {
+      case SIGCHLD:
```

Comment:
You can remove the special handling of these three signals. We used to have that in the Linux implementation but there's no use for it.